plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcClient.java

```diff
@@ -94,6 +95,8 @@ PreparedStatement buildSql(ConnectorSession session, Connection connection, Jdbc
 
     boolean supportsLimit();
 
+    boolean supportsTopNPushDown();
```

Comment:
It can become more tricky than simple yes/no. For example, it may depend on actual sort items.
I think you can just remove this method and make `prepareTopNQuery` return an `Optional`